tests/hpc/ganglia_client.pm

```diff
@@ -39,18 +47,17 @@ sub run {
     for (1 .. $max_retries) {
         eval {
             # Check if gmond has connected to gmetad
-            validate_script_output "gstat -a", sub { m/.*Hosts: 2.*/ };
+            validate_script_output "gstat -a", sub { m/.*Hosts: ${nodes}.*/ }, 0;
         };
         last unless ($@);
         record_info 'waiting for nodes', 'Not all nodes connected yet. Retrying...';
     }
     die "Not all nodes were connected after $max_retries retries." if $@;
 
-
     # Check if an arbitrary value could be sent via gmetric command
     my $testMetric = "openQA";
     type_string "gmetric -n \"$testMetric\" -v \"openQA\" -t string | tee /dev/ttyS0";
-    assert_script_run "echo \"\\n\" | nc $server_ip 8649 | grep $testMetric";
+    assert_script_run "echo \"\\n\" | nc ganglia-server 8649 | grep $testMetric";
```

Comment:
this means that if hostname would differ from "ganglia-server" we will have test failure , let's add some explicit check and failure with clear error message in ganglia_server.pm for hostname.

Reply:
@asmorodskyi code was updated with variable `GANGLIA_SERVER_HOSTNAME`